### PR TITLE
`prefer-number-properties`: Add `checkNaN` option

### PR DESCRIPTION
**Pull request by gurgunday**
I think the `NaN` check should also be toggleable, because I use it like I use `null` (absence of an object), to represent an absence of a number; and since `null` is global, this setting lets me be more consistent

Thanks for maintaining this amazing plugin!

### CODE REVIEW

**Comment by gurgunday** [comment]
CI error not related I believe

**Comment by sindresorhus** [comment]
// @Richienb

**Comment by fisker** [comment]
Should the option be `checkNaN` or `checkNan` ?

**Comment by Richienb** [comment]
Could we maybe set it to false by default? (breaking)
I also think that `NaN` is better than `Number.NaN`.

**Comment by sindresorhus** [comment]
> Could we maybe set it to false by default? (breaking)

👍

File: docs/rules/prefer-number-properties.md [hunk 1]
```diff
@@ -93,6 +93,13 @@ Default: `false`
 
 Pass `checkInfinity: true` to enable check on `Infinity`.
 
+### checkNaN
+
+Type: `boolean`\
+Default: `true`
+
+Pass `checkNaN: false` to disable check on `NaN`.
```
Comment (Richienb) [inline docs/rules/prefer-number-properties.md:101]:
```suggestion
Default: `false`

Pass `checkNaN: true` to enable check on `NaN`.
```

**Review by Richienb — CHANGES_REQUESTED** [review]

File: docs/rules/prefer-number-properties.md (hunk 1 above)
Comment (gurgunday) [reply docs/rules/prefer-number-properties.md:101]: Yes, there are also a number of tests that need to be modified (ran with checkNaN: true), will do them

File: docs/rules/prefer-number-properties.md [hunk 2]
```diff
@@ -93,6 +93,13 @@ Default: `false`
...
 Pass `checkInfinity: true` to enable check on `Infinity`.
 
+### checkNaN
+
+Type: `boolean`\
+Default: `false`
+
+Pass `checkNaN: true` to enable check on `NaN`.
+
 #### Fail
```
Comment (silverwind) [inline docs/rules/prefer-number-properties.md:103]: Your docs change is in the wrong place, the `Fail` section below is for `checkInfinity`, so you should at it below and add your own `Fail` examples.

**Comment by gurgunday** [comment]
Yeah, sorry y'all — when I turn it off by default, a bunch of tests are failing and even though I modify them to be like Infinity checks, there are errors

I do not have the time to look into it, so let's focus on adding this feature in this PR

See f544b63a4221e51fd5dc30fbf8e1c2027bfb502e to view my latest try

**Review by Richienb — CHANGES_REQUESTED** [review]
I would like some examples in the docs

**Comment by Richienb** [comment]
> I do not have the time to look into it, so let's focus on adding this feature in this PR

That's ok, we can change it in the next PR.

Added to the tracking issue

**Comment by gurgunday** [comment]
I added one more, what else would you like to see?

File: docs/rules/prefer-number-properties.md [hunk 3]
```diff
@@ -126,3 +126,39 @@ const isPositiveZero = value => value === 0 && 1 / value === Number.POSITIVE_INF
...
+const foo = NaN;
+```
+
+```js
+// eslint unicorn/prefer-number-properties: ["error", {"checkNaN": false}]
+const foo = -NaN;
+```
+
+```js
+// eslint unicorn/prefer-number-properties: ["error", {"checkNaN": true}]
```
Comment (Richienb) [inline docs/rules/prefer-number-properties.md:162]: Add cases that specifically show that `Number.NaN` and `-Number.NaN` pass
I don't think this example is necessary.

**Review by Richienb — CHANGES_REQUESTED** [review]

File: docs/rules/prefer-number-properties.md (hunk 3 above)
Comment (gurgunday) [reply docs/rules/prefer-number-properties.md:162]: Oh sorry; I was confused because Inifinity doesn't have examples like that either

**Review by Richienb — APPROVED** [review]

**Review by sindresorhus — APPROVED** [review]

**Pull request merged**